### PR TITLE
Make document currentscript work

### PR DESCRIPTION
**Pull request by dralletje**
Instead of eval-ing the code from javascript, I now insert a new script element that gets actually executed (so it becomes the `document.currentScript`). It won't keep `currentScript` after anything async, but that also doesn't work with normal scripts either, so I think this just raises compatibility a bunch.

Maybe even deprecate naked `currentScript` in favor of `document.currentScript`? but I don't care much either way.

Supersedes #1522

### CODE REVIEW

**Comment by fonsp** [comment]
Uh oh, I think I wrote some scripts that use `currentScript` in async stuff

**Comment by dralletje** [comment]
Yeah so we might first want to deprecate it if we remove it. You can still do `let currentScript = window.currentScript`.

But this PR doesn't remove `currentScript`.

File: frontend/components/CellOutput.js [hunk 1]
```diff
@@ -282,16 +306,19 @@ const execute_scripttags = async ({ root_node, script_nodes, previous_results_ma
...
-                            invalidation: invalidation,
-                            getPublishedObject: (id) => cell.getPublishedObject(id),
-                            ...observablehq_for_cells,
-                        },
-                        code: node.innerText,
+                    let { node: new_node, result } = await execute_inside_script_tag_that_replace(node, async () => {
+                        return await execute_dynamic_function({
+                            environment: {
+                                this: script_id ? old_result : window,
+                                currentScript: document.currentScript,
```
Comment (fonsp) [inline frontend/components/CellOutput.js:313]: if we leave this at `node`, it will still work in async things, right?

Comment (fonsp) [reply frontend/components/CellOutput.js:313]: Because right now it would break async stuff?

Comment (dralletje) [reply frontend/components/CellOutput.js:313]: Sorry for the confusion, the "might remove currentScript" in my original message was a suggestion. The `currentScript` variable works exactly as it did before, with and without async.

Comment (fonsp) [reply frontend/components/CellOutput.js:313]: Nice! I guess i should just try it out

**Comment by fonsp** [comment]
Can you add some comments to the code? It's too difficult :)

Start by copy pasting your PR description into a docstring?

**Review by fonsp — CHANGES_REQUESTED** [review]
Can you add some comments to the code? It's too difficult :)

Start by copy pasting your PR description into a docstring?

**Comment by dralletje** [comment]
@fonsp 😏

**Comment by fonsp** [comment]
All the crazy things I could come up with still work, awesome!

```julia
# ╔═╡ b474a054-846a-11ec-0f51-1b8e8c589aae
using HypertextLiteral

# ╔═╡ 3b09d217-42b9-473c-b98c-bcbbc1c856a4
@htl """
<span>
	<button>click meeee</button>
	<script>
		const span = currentScript.parentElement
		const btn = span.querySelector("button")

		btn.onclick = () => {
			console.log(currentScript, currentScript.parentElement, document.currentScript)
		}
	</script>
</span>


"""

# ╔═╡ 6868ed5d-0ca4-4a1a-be17-e8bde8f0f8ef
zz

# ╔═╡ adbcbc43-c88a-469b-8e8b-49a20706b415
@bind zz @htl """<script>
	let value = 1

	let handler = setInterval(() => {
		value += 1
		currentScript.value = value
		currentScript.dispatchEvent(
			new CustomEvent("input")
		)
	}, 500)

	invalidation.then(() => {
		clearInterval(handler)
	})
	
</script>
"""
```

**Comment by cjdoris** [comment]
@dralletje Thanks so much for this! Can confirm that PythonCall objects (such as Altair plots) now display correctly in Pluto 0.18.0.

**Pull request merged**